### PR TITLE
Migrate Realm Models to Multiplatform common library

### PR DESCRIPTION
**Pull request by TNychka**

### CODE REVIEW

File: Habitica/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -7,5 +7,6 @@
     </debug-overrides>
     <domain-config cleartextTrafficPermitted="true">
         <domain includeSubdomains="true">192.168.178.52</domain>
+        <domain includeSubdomains="true">10.0.2.2</domain>
```
Comment (TNychka) [inline Habitica/res/xml/network_security_config.xml:10]: Necessary component of local testing.

File: Habitica/src/main/java/com/habitrpg/android/habitica/HabiticaBaseApplication.kt [hunk 2]
```diff
@@ -110,6 +111,7 @@ abstract class HabiticaBaseApplication : MultiDexApplication() {
         Realm.init(this)
         val builder = RealmConfiguration.Builder()
                 .schemaVersion(1)
+                .modules(Realm.getDefaultModule()!!, MultiplatformRealmLibrary())
```
Comment (TNychka) [inline Habitica/src/main/java/com/habitrpg/android/habitica/HabiticaBaseApplication.kt:114]: Enables realm to pick up the new shared library

File: shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/nativePackages/NativeColor.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package com.habitrpg.shared.habitica.nativePackages
```
Comment (TNychka) [inline shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/nativePackages/NativeColor.kt:1]: I don't really like the way that this has been done. I want to investigate this more, but for now it works

File: Habitica/src/main/java/com/habitrpg/android/habitica/utils/PurchasedDeserializer.java [hunk 4]
```diff
@@ -68,7 +68,7 @@ public Purchases deserialize(JsonElement json, Type typeOfT, JsonDeserialization
             }
         }
 
-        purchases.customizations = customizations;
+        purchases.setCustomizations(customizations);
```
Comment (TNychka) [inline Habitica/src/main/java/com/habitrpg/android/habitica/utils/PurchasedDeserializer.java:71]: Result of converting purchases to kotlin from java

File: shared/src/commonMain/kotlin/com/habitrpg/shared/habitica/models/Tag.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    var tasks: NativeList<Task>? = null
+    var name: String = ""
+    internal var challenge: Boolean = false
+
+    fun getTasks(): List<Task>? {
+        return tasks
+    }
+
+    override fun equals(o: Any?): Boolean {
+        if (o is Tag) {
```
Comment (TNychka) [inline shared/src/commonMain/kotlin/com/habitrpg/shared/habitica/models/Tag.kt:23]: Need to verify that this works the same as        ` if (Tag::class.java.isAssignableFrom(o!!.javaClass)) {`

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/helpers/TaskTextParser.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+object TaskTextParser {
+    fun parseMarkdown(task: Task?) {
+        task?.parsedText = MarkdownParser.parseMarkdown(task?.text)
+        task?.parsedNotes = MarkdownParser.parseMarkdown(task?.notes)
+    }
+
+    fun markdownText(task: Task, callback: (CharSequence) -> Unit): CharSequence {
+        if (task.parsedText != null) {
+            return task.parsedText ?: ""
```
Comment (hulien22) [inline Habitica/src/main/java/com/habitrpg/android/habitica/ui/helpers/TaskTextParser.kt:14]: is the elvis operator needed here?

**Review by hulien22 — APPROVED** [review]
LGTM :rocket:

Should really add a bit more to the description here just so that's it easier to review ;)

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/helpers/TaskTextParser.kt (hunk 6 above)
Comment (TNychka) [reply Habitica/src/main/java/com/habitrpg/android/habitica/ui/helpers/TaskTextParser.kt:14]: Not needed, good catch!

File: shared/src/commonMain/kotlin/com/habitrpg/shared/habitica/models/Tag.kt (hunk 5 above)
Comment (TNychka) [reply shared/src/commonMain/kotlin/com/habitrpg/shared/habitica/models/Tag.kt:23]: Basic debugging shows that this works, however test cases should be added

**Pull request closed without merging**